src/utils/hooks/useLanguage.ts

```diff
@@ -0,0 +1,94 @@
+import { usePathname } from "next/navigation";
+import { useEffect, useState } from "react";
+
+import { fetchWithToken } from "studio/lib/fetchWithToken";
+import { SlugTranslations } from "studio/lib/interfaces/slugTranslations";
+import { LanguageObject } from "studio/lib/interfaces/supportedLanguages";
+import { LANGUAGES_QUERY } from "studio/lib/queries/languages";
+import { SLUG_TRANSLATIONS_FROM_LANGUAGE_QUERY } from "studio/lib/queries/slugTranslations";
+
+function useSlugTranslations(
+  currentLanguage: LanguageObject | undefined,
+  slug: string,
+  availableLanguages: LanguageObject[] | null,
+) {
+  const [slugTranslationsData, setSlugTranslationsData] =
+    useState<SlugTranslations | null>(null);
+
+  useEffect(() => {
+    if (currentLanguage === undefined) {
+      return;
+    }
+    fetchWithToken<SlugTranslations | null>(
+      SLUG_TRANSLATIONS_FROM_LANGUAGE_QUERY,
+      {
+        slug,
+        language: currentLanguage?.id,
+      },
+    ).then((data) => setSlugTranslationsData(data));
```

Comment:
I thiiiink you can just do? 

```suggestion
    ).then(setSlugTranslationsData);
```